BackEnd/src/main/kotlin/quest/marketstack/tradingapp/service/opentrades/OpenTradeService.kt

```diff
@@ -20,40 +23,42 @@ class OpenTradeService(private val dataSource: OpenTradeDataSource,private val c
         var isMatch = false
         for (i in execList) {
             openTrades = dataSource.retrieveTrades()
-            isMatch=false
+            isMatch = false
             for (j in openTrades) {
                 val tradeId = j.id
                 if (i.symbol == j.tradeExecs.first().symbol && tradeId != null) {
-                    if ((j.shortLong&&(i.side=="SS"||i.side=="BC"))||(!j.shortLong&&(i.side=="B"||i.side=="S"))){
+                    if ((j.shortLong && (i.side == "SS" || i.side == "BC")) || (!j.shortLong && (i.side == "B" || i.side == "S"))) {
                         continue
                     }
                     print("Execution added to existing trade")
                     dataSource.addExec(i, tradeId)
                     isMatch = true
                 }
             }
-            if (!isMatch){
+            if (!isMatch) {
                 print("New Trade Opened")
-                if (i.side == "B")
+                if (i.side == "B") {
                     dataSource.createTrades(listOf(OpenTrade(tradeExecs = mutableListOf(i), shortLong = true)))
-                if (i.side == "SS")
+                }
+                if (i.side == "SS") {
                     dataSource.createTrades(listOf(OpenTrade(tradeExecs = mutableListOf(i), shortLong = false)))
+                }
             }
-
         }
```

Comment:
The nested for loop here is going to slow down your application a lot. There really is no reason to be using for loops in professional code. 

To point you in the right direction, here is some Kotlin info (if you already know this I apologize): Collection is the root (parent) class of all of the Iterable Kotlin objects. So, Collection is the parent class of List, Set, Map, and ArrayDeque. Look at this link: https://kotlinlang.org/docs/collections-overview.html#collection

There really should be no situation where you want to use a collection. You're using the parent class of a List or Set or Map etc, so you're just using a way to store data that has _less_ functionality for really no reason. Go back through your code and find everywhere that you're using a Collection, and figure out which of the children you should use instead. As a hint, you really should only ever need to use Map and List. 

Replace all the instances of `Collection` with `Map` or `List` depending on what you think makes the most sense. Let me know if you need more explanation/need help